gdal2 MDB configure error

When trying to build gdal2 `--with-mdb` I get a configure error:
```
checking whether we should include MDB support... configure: error: "--with-jvm-lib must be specified."
```

--- a/Formula/gdal2-mdb.rb
+++ b/Formula/gdal2-mdb.rb
@@ -56,7 +56,7 @@ class Gdal2Mdb < Formula
       "--with-macosx-framework",
     ]
 
-    cmd = Language::Java.java_home_cmd("1.8+")
+    cmd = Language::Java.java_home_cmd("1.8")
     ENV["JAVA_HOME"] = Utils.popen_read(cmd).chomp
     args << "--with-java=yes"
     args << "--with-jvm-lib=#{ENV["JAVA_HOME"]}/jre/lib/server"

--- a/Formula/gdal2.rb
+++ b/Formula/gdal2.rb
@@ -40,6 +40,7 @@ class Gdal2 < Formula
   deprecated_option "enable-unsupported" => "with-unsupported"
   deprecated_option "enable-mdb" => "with-mdb"
   deprecated_option "complete" => "with-complete"
+  deprecated_option "with-java" => "with-swig-java"
 
   depends_on "libpng"
   depends_on "jpeg"
@@ -84,9 +85,8 @@ class Gdal2 < Formula
     depends_on "json-c"
   end
 
-  depends_on :java => ["1.7+", :optional, :build]
-
   if build.with? "swig-java"
+    depends_on :java => ["1.8", :build]
     depends_on "ant" => :build
     depends_on "swig" => :build
   end
@@ -274,9 +274,11 @@ class Gdal2 < Formula
     include.install Dir["gnm/**/*.h"] if build.with? "gnm"
 
     if build.with? "swig-java"
+      cmd = Language::Java.java_home_cmd("1.8")
+      ENV["JAVA_HOME"] = Utils.popen_read(cmd).chomp
       cd "swig/java" do
         inreplace "java.opt", "linux", "darwin"
-        inreplace "java.opt", "#JAVA_HOME = /usr/lib/jvm/java-6-openjdk/", "JAVA_HOME=$(shell echo $$JAVA_HOME)"
+        inreplace "java.opt", "#JAVA_HOME = /usr/lib/jvm/java-6-openjdk/", "JAVA_HOME=#{ENV["JAVA_HOME"]}"
         system "make"
         system "make", "install"
 
@@ -313,6 +315,8 @@ class Gdal2 < Formula
 
     if build.with? "mdb"
       s += <<~EOS
+
+      MDB is no longer supported in the main GDAL formula, please use gdal2-mdb instead.
 
       To have a functional MDB driver, install supporting .jar files in:
         `/Library/Java/Extensions/`

--- a/travis/before_deploy.sh
+++ b/travis/before_deploy.sh
@@ -50,6 +50,7 @@ fi
 
 
 # Build the bottles
+BUILT_BOTTLES=
 mkdir -p bottles
 
 pushd bottles
@@ -70,14 +71,24 @@ pushd bottles
 
 #    Do Merge bottles with the formula
 #    Don't commit anything, we'll do that after updating all the formulae
-    brew bottle --merge --write --no-commit ${f}*.json
+#    Catch the eror and store it to a variable
+    {bottle_error=$(brew bottle --merge --write --no-commit ${f}*.json 2>&1 1>&3) ;} 3>&1
+#   If there's an error, remove the json and bottle files, we don't want them anymore.
+    if [[ -n $bottle_error ]]; then
+      echo "Unable bottle ${f}"
+      echo $bottle_error
+      rm ${f}*.json
+      rm ${f}*.tar.gz
+    else
+      BUILT_BOTTLES="$BUILT_BOTTLES ${f}"
+    fi
   done
 popd
 
 # Now do the commit and push
 
 git add -vA Formula/*.rb
-git commit -m "Updated bottles for: ${CHANGED_FORMULAE}
+git commit -m "Updated bottles for: ${BUILT_BOTTLES}
 
 Committed for ${COMMIT_USER}<${COMMIT_EMAIL}>
 [ci skip]"

